src/libfsm/determinise.c

```diff
@@ -292,6 +285,129 @@ hash_single_state(const void *a)
 	return hashrec(st, sizeof *st);
 }
 
+/* builds transitions of the glushkov NFA
+ *
+ * Makes a single pass through each of the nfa states that the glushkov nfa
+ * state corresponds to and constructs an edge set for each symbol out of the
+ * nfa states.
+ *
+ * It does this in one pass by maintaining an array of destination states where
+ * the index of the array is the transition symbol.
+ */
+static int
+glushkov_buildtransitions(const struct fsm *fsm, struct fsm *dfa, struct mapping_set *mappings, struct mapping *curr)
+{
+	struct fsm_state *s;
+	struct state_iter it;
+	int sym, sym_min, sym_max;
+	int ret = 0;
+
+	struct state_array outedges[FSM_SIGMA_COUNT];
+	struct hashset *outsets[FSM_SIGMA_COUNT];
+
+	assert(fsm != NULL);
+	assert(dfa != NULL);
+	assert(curr != NULL);
+	assert(curr->closure != NULL);
+
+	sym_min = UCHAR_MAX+1;
+	sym_max = -1;
```

Comment:
Rather than `UCHAR_MAX+1`, better to say `FSM_SIGMA_MAX`, if you're after an out-of-bounds value one past the maximum symbol in the alphabet.

Reply:
done: a32ef76